hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/util/PartitionFilterGenerator.java

```diff
@@ -0,0 +1,212 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.hive.util;
+
+import org.apache.hudi.common.util.ReflectionUtils;
+import org.apache.hudi.common.util.collection.Pair;
+import org.apache.hudi.hive.HiveSyncConfig;
+import org.apache.hudi.hive.HoodieHiveSyncException;
+import org.apache.hudi.hive.expression.And;
+import org.apache.hudi.hive.expression.AttributeReferenceExpression;
+import org.apache.hudi.hive.expression.EqualTo;
+import org.apache.hudi.hive.expression.Expression;
+import org.apache.hudi.hive.expression.GreatThanOrEqual;
+import org.apache.hudi.hive.expression.LessThanOrEqual;
+import org.apache.hudi.hive.expression.Literal;
+import org.apache.hudi.hive.expression.Or;
+import org.apache.hudi.sync.common.model.FieldSchema;
+import org.apache.hudi.sync.common.model.Partition;
+import org.apache.hudi.sync.common.model.PartitionValueExtractor;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.apache.hudi.hive.HiveSyncConfig.HIVE_SYNC_FILTER_PUSHDOWN_MAX_SIZE;
+import static org.apache.hudi.sync.common.HoodieSyncConfig.META_SYNC_PARTITION_EXTRACTOR_CLASS;
+
+public class PartitionFilterGenerator {
+
+  private static final Set<String> SUPPORT_TYPES = new HashSet<String>() {
+    {
+      add(HiveSchemaUtil.INT_TYPE_NAME);
+      add(HiveSchemaUtil.BIGINT_TYPE_NAME);
+      add(HiveSchemaUtil.DATE_TYPE_NAME);
+      add(HiveSchemaUtil.STRING_TYPE_NAME);
+    }
+  };
+
+  /**
+   * Build expression from the Partition list. Here we're trying to match all partitions.
+   *
+   * ex. partitionSchema(date, hour) [Partition(2022-09-01, 12), Partition(2022-09-02, 13)] =>
+   *     Or(And(Equal(Attribute(date), Literal(2022-09-01)), Equal(Attribute(hour), Literal(12))),
+   *     And(Equal(Attribute(date), Literal(2022-09-02)), Equal(Attribute(hour), Literal(13))))
+   */
+  private static Expression buildPartitionExpression(List<Partition> partitions, List<FieldSchema> partitionFields) {
+    return partitions.stream().map(partition -> {
+      List<String> partitionValues = partition.getValues();
+      Expression root = null;
+
+      for (int i = 0; i < partitionFields.size(); i++) {
+        FieldSchema field = partitionFields.get(i);
+        String value = partitionValues.get(i);
+        EqualTo exp = new EqualTo(new AttributeReferenceExpression(field.getName()),
+            new Literal(value, field.getType()));
+        if (root != null) {
+          root = new And(root, exp);
```

Comment:
@boneanxs i'd suggest to instead of creating separate classes just do static methods in the BinaryOperator like following:

```
BinaryOperator.and(left, right);
BinaryOperator.gteq(left, right);
// ...
```

This will make it easier for us to reconcile this in the near future w/ RFC-64 (having individual classes compounds the overhead of handling them)